Split UI how-to guides into per-platform guides (#781)
THEOplayer/documentation: theoplayer/how-to-guides/roku/ui/remove-ui-element.mdx

---
sidebar_position: 2
sidebar_label: Remove an element from the UI
---

# How to remove an element from the UI on Roku

import Intro from '../../shared/ui/_remove-ui-element-intro.mdx';

<Intro />

## Implementation

> What UI elements should be visible for Chromefull THEOplayer on Roku? Currently, Chromefull THEOplayer uses Roku native player and that one has e.g. a clock UI element while paused.

Reviewer: ```suggestion
To control which UI elements should be visible for the THEOplayer on Roku? Currently, THEOplayer uses the Roku native player and its controls by default. To remove an element from these controls, you can remove the elements manually. For example, the default controls have a clock UI element while paused. To remove it, follow the code below:
```

```xml
<?xml version="1.0" encoding="utf-8" ?>
<component name="TestScene" extends="Scene">
    <interface>
    </interface>

    <script type = "text/brightscript" >

        <![CDATA[

        sub Init()
            m.player = m.top.findNode("TestPlayer")
            m.player.configuration = {

>               "license": "" ' put the THEOplayer license between apostrophes

Reviewer: To follow the convention used elsewhere, let's put in a token license.
```suggestion
              "license": "<YOUR_LICENSE_STRING>"
```

            }
            ' remove the clock UI node

>             m.player.getChildren(-1, 0)[0].getChildren(-1, 0)[1].removeChildIndex(11)

Reviewer: (aside) unorthodox, but ok, that theoretically works

            setSource()
            m.player.setFocus(true)
            m.player.callFunc("play")
        end sub

        sub setSource()

>             sourceDescription = {
>                 "sources": [
>                     {
>                         "src": "http://cdn.theoplayer.com/video/star_wars_episode_vii-the_force_awakens_official_comic-con_2015_reel_(2015)/index.m3u8",
>                         "type": "application/x-mpegURL"
>                     }
>                 ]
>             }
>             m.player.source = sourceDescription
>             m.player.source.Live = false
>             m.player.source.LiveBoundsPauseBehavior = "pause"
>         end sub

Reviewer: Setting those properties after the source is set isn't going to work reliably. We also don't allow the client to set LiveBoundsPauseBehavior currently.
```suggestion
        sub setSource()
            sourceDescription = {
                "sources": [
                    {
                        "src": "http://cdn.theoplayer.com/video/star_wars_episode_vii-the_force_awakens_official_comic-con_2015_reel_(2015)/index.m3u8",
                        "live": true,
                        "type": "application/x-mpegURL"
                    }
                ]
            }
            m.player.source = sourceDescription
        end sub
```

        ]]>

    </script>

    <children>
	    <THEOsdk:THEOplayer id="TestPlayer"/>
    </children>
</component>

> ```

Reviewer: Devin, please add an example below that also shows how to turn all default UI off by setting the THEOplayer's `controls` field to `false`.
